File: benchmarking/GenSSI/Akt-pathway.m
function model = SMTH()
syms EGF_EGFR EGFR pEGFR pAkt_S6 pEGFR_Akt pAkt S6 pS6 Akt
syms a2 a3 reaction_1_k1 reaction_4_k1 reaction_9_k1 reaction_1_k2 reaction_7_k1 reaction_6_k1 a1 reaction_3_k1 reaction_5_k1 reaction_2_k1 EGFR_turnover reaction_2_k2 reaction_5_k2 reaction_8_k1
syms EGF_EGFR0 EGFR0 pEGFR0 pAkt_S60 pEGFR_Akt0 pAkt0 S60 pS60 Akt0
syms pro_EGFR
model.sym.p = [a2; a3; reaction_1_k1; reaction_4_k1; reaction_9_k1; reaction_1_k2; reaction_7_k1; reaction_6_k1; a1; reaction_3_k1; reaction_5_k1; reaction_2_k1; EGFR_turnover; reaction_2_k2; reaction_5_k2; reaction_8_k1; EGF_EGFR0; EGFR0; pEGFR0; pAkt_S60; pEGFR_Akt0; pAkt0; S60; pS60; Akt0];
model.sym.x = [EGF_EGFR; EGFR; pEGFR; pAkt_S6; pEGFR_Akt; pAkt; S6; pS6; Akt];
model.sym.g = [pro_EGFR];
model.sym.x0 = [EGF_EGFR0; EGFR0; pEGFR0; pAkt_S60; pEGFR_Akt0; pAkt0; S60; pS60; Akt0];
model.sym.xdot = [reaction_1_k1*EGF_EGFR - reaction_9_k1*EGF_EGFR - reaction_1_k2*EGF_EGFR
-reaction_1_k1*EGF_EGFR + reaction_1_k2*EGF_EGFR - EGFR*EGFR_turnover + EGFR_turnover*pro_EGFR
-reaction_4_k1*pEGFR + reaction_9_k1*EGF_EGFR - pEGFR*Akt*reaction_2_k1 + reaction_3_k1*pEGFR_Akt + pEGFR_Akt*reaction_2_k2
pAkt*reaction_5_k1*S6 - reaction_6_k1*pAkt_S6 - pAkt_S6*reaction_5_k2
pEGFR*Akt*reaction_2_k1 - reaction_3_k1*pEGFR_Akt - pEGFR_Akt*reaction_2_k2
-pAkt*reaction_7_k1 - pAkt*reaction_5_k1*S6 + reaction_6_k1*pAkt_S6 + reaction_3_k1*pEGFR_Akt + pAkt_S6*reaction_5_k2
pS6*reaction_8_k1 - pAkt*reaction_5_k1*S6 + pAkt_S6*reaction_5_k2
-pS6*reaction_8_k1 + reaction_6_k1*pAkt_S6
pAkt*reaction_7_k1 - pEGFR*Akt*reaction_2_k1 + pEGFR_Akt*reaction_2_k2];
model.sym.y = [pEGFR*a1 + a1*pEGFR_Akt
a2*pAkt + a2*pAkt_S6
pS6*a3];
end

File: benchmarking/IdentifiableFunctions/systems/Akt pathway/Akt pathway.jl
# Akt pathway
#! format: off
using StructuralIdentifiability

system = @ODEmodel(
	EGFR'(t) = -reaction_1_k1*EGF_EGFR(t) + reaction_1_k2*EGF_EGFR(t) - EGFR(t)*EGFR_turnover + EGFR_turnover*pro_EGFR(t),
	pAkt'(t) = -pAkt(t)*reaction_7_k1 - pAkt(t)*reaction_5_k1*S6(t) + reaction_6_k1*pAkt_S6(t) + reaction_3_k1*pEGFR_Akt(t) + pAkt_S6(t)*reaction_5_k2,
	pEGFR_Akt'(t) = pEGFR(t)*Akt(t)*reaction_2_k1 - reaction_3_k1*pEGFR_Akt(t) - pEGFR_Akt(t)*reaction_2_k2,
	S6'(t) = pS6(t)*reaction_8_k1 - pAkt(t)*reaction_5_k1*S6(t) + pAkt_S6(t)*reaction_5_k2,
	pEGFR'(t) = -reaction_4_k1*pEGFR(t) + reaction_9_k1*EGF_EGFR(t) - pEGFR(t)*Akt(t)*reaction_2_k1 + reaction_3_k1*pEGFR_Akt(t) + pEGFR_Akt(t)*reaction_2_k2,
	EGF_EGFR'(t) = reaction_1_k1*EGF_EGFR(t) - reaction_9_k1*EGF_EGFR(t) - reaction_1_k2*EGF_EGFR(t),
	Akt'(t) = pAkt(t)*reaction_7_k1 - pEGFR(t)*Akt(t)*reaction_2_k1 + pEGFR_Akt(t)*reaction_2_k2,
	pAkt_S6'(t) = pAkt(t)*reaction_5_k1*S6(t) - reaction_6_k1*pAkt_S6(t) - pAkt_S6(t)*reaction_5_k2,
	pS6'(t) = -pS6(t)*reaction_8_k1 + reaction_6_k1*pAkt_S6(t),
	y1(t) = pEGFR(t)*a1 + a1*pEGFR_Akt(t),
	y2(t) = a2*pAkt(t) + a2*pAkt_S6(t),
	y3(t) = pS6(t)*a3
)
